smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/member/MemberShapeDecodeGenerator.kt

```diff
@@ -290,4 +293,6 @@ open class MemberShapeDecodeGenerator(
             )
         }
     }
+
+    private var readingRequiredValue: Boolean = shapeContainingMembers.isUnionShape
```

Comment:
Extracted `readingRequiredValue` to a helper.  In the future, if nonoptional values are supported, this logic may become more complex.